File: tests/fixtures/sample.go
package main

import "fmt"

type Greeter struct {
	Name string
}

func (g *Greeter) Greet() string {
	return fmt.Sprintf("Hello, %s!", g.Name)
}

func helper(x int) int {
	return x * 2
}

func main() {
	g := &Greeter{Name: "World"}
	msg := g.Greet()
	fmt.Println(msg)
	result := helper(42)
	fmt.Println(result)
}

File: tests/fixtures/sample.js
const fs = require('fs');

class Logger {
    constructor(prefix) {
        this.prefix = prefix;
    }

    log(message) {
        console.log(`${this.prefix}: ${message}`);
    }
}

function greet(name) {
    return `Hello, ${name}!`;
}

function main() {
    const logger = new Logger("App");
    const msg = greet("World");
    logger.log(msg);
}

main();
